aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -415,15 +415,19 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
             .tableType(fromTable.tableType())
             .parameters(fromTable.parameters())
             .storageDescriptor(fromTable.storageDescriptor());
-
-    glue.createTable(
-        CreateTableRequest.builder()
-            .catalogId(awsProperties.glueCatalogId())
-            .databaseName(toTableDbName)
-            .tableInput(tableInputBuilder.name(toTableName).build())
-            .build());
-    LOG.info("created rename destination table {}", to);
-
+    try {
+        glue.createTable(
+            CreateTableRequest.builder()
+                .catalogId(awsProperties.glueCatalogId())
+                .databaseName(toTableDbName)
+                .tableInput(tableInputBuilder.name(toTableName).build())
+                .build());
+        LOG.info("created rename destination table {}", to);
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+    // Catch AWS Glue exception and rethrow it as Iceberg's AlreadyExistsException
+    throw new org.apache.iceberg.exceptions.AlreadyExistsException("Table already exists: " + toTableName, e);
```

Comment:
Can we use the fully qualified `AlreadyExistsException` for glue and then just use the normal class name the Iceberg `AlreadyExistsException`. That aligns with what's done in the other parts of the code and is a bit more readable imo.